### DISCUSSION TITLE
如何微调 en_PPOCRv4 模型以添加更多角色 How to fine tune en_PPOCRv4 model to add more characters.

### DISCUSSION THREAD

**Discussion by CapturefastCEO** [PaddlePaddle/PaddleOCR#13754] [Q&A]
我想用八个新字符对 en_PPOCRv4 模式进行微调，使其覆盖其原始词典。
我在 en_dict.txt 末尾添加了新字符。
我准备了只有新字符的单词的数据集。然后我通过 en_PP-OCRv4_rec 训练了模型。
我对新字符的结果感到很满意。但是，最新模型无法识别字母 X、Q 和数字，因为数据集中没有包含这些字母的单词。它忘记了之前的功能。

所以，看起来我做错了什么，但是什么呢？我希望新模型能够理想地识别之前所做的一切并开始识别新单词。
如何在不失去先前识别能力的情况下在训练过的模型上添加更多字符？

I want to fine-tune en_PPOCRv4 mode with eight new characters over its original dictionary. 
I added the new characters at the end of en_dict.txt.
I prepared the dataset with words that only had new characters. And I trained the model over en_PP-OCRv4_rec.
I'm delighted with the results of the new characters. However, the latest model does not recognize the letters X, Q, and numbers since the dataset does not have words with these letters. It forgets its previous capabilities. 

So, it looks like I'm doing something wrong, but what?   I expected the new model to ideally recognize everything it did before and start recognizing the new words. 
How can I add more characters over a trained model without losing the previous recognition capabilities? 
Thank ln advance

**Comment by CapturefastCEO** [comment]
I have never received an answer to my questions until today. Is there a rule regarding posting questions about PaddleOCR?

**Comment by GreatV** [comment]
please refer to: https://paddlepaddle.github.io/PaddleOCR/en/ppocr/model_train/training.html

**Reply by GreatV** [reply]
Instead of training the model only on the new characters, create a mixed dataset that includes both the new characters and the original ones (including X, Q, and numbers).
Ensure that the dataset has a balanced distribution of samples containing both old and new characters. This will help the model retain its previous knowledge while learning the new characters.

**Reply by CapturefastCEO** [reply]
Hi GreatV. I read the link, but I still have a lot of questions.
How will I know how many image samples should I use for each new character? 
I first trained the model with 2000 images and had great results but lost the spaces. Then, I used 20,000 samples, which took two days to train the model. I got back the spaces but got very poor results on capital letters. 
How will I know the best balance?
Thanks
